packages/notebook/src/browser/view-model/notebook-model.ts

```diff
@@ -170,9 +163,30 @@ export class NotebookModel implements Saveable, Disposable {
         };
     }
 
+    async applySnapshot(snapshot: Saveable.Snapshot): Promise<void> {
+        const data = 'read' in snapshot ? JSON.parse(snapshot.read()!) : JSON.parse(snapshot.value);
```

Comment:
Suggestion: We shouldn't assume that `snapshot.read()` returned a non-null value. We just just return from the method in case anything is `null` or `undefined` here.